climada/hazard/tc_tracks.py

```diff
@@ -3094,3 +3094,196 @@ def _zlib_from_dataarray(data_var: xr.DataArray) -> bool:
+        )
+
+    # define grid resolution and bounds for density computation
+    if not bounds:
+        lon_min, lat_min, lon_max, lat_max = -180, -90, 180, 90
+    else:
+        lon_min, lat_min, lon_max, lat_max = bounds[0], bounds[1], bounds[2], bounds[3]
+
+    lat_bins: np.ndarray = np.linspace(lat_min, lat_max, int(180 / res))
+    lon_bins: np.ndarray = np.linspace(lon_min, lon_max, int(360 / res))
+
+    # compute 2D density
+    if genesis:
+        hist_count = compute_genesis_density(
+            tc_track=tc_track, lat_bins=lat_bins, lon_bins=lon_bins
+        )
+    else:
+        hist_count = np.zeros((len(lat_bins) - 1, len(lon_bins) - 1))
+        for track in tqdm(tc_track.data, desc="Processing Tracks"):
+
+            # select according to wind speed
+            wind_speed = track.max_sustained_wind.values
+            if wind_min and wind_max:
+                index = np.where((wind_speed >= wind_min) & (wind_speed < wind_max))[0]
+            elif wind_min and not wind_max:
+                index = np.where(wind_speed >= wind_min)[0]
+            elif wind_max and not wind_min:
+                index = np.where(wind_speed <= wind_max)[0]
+            else:
+                index = slice(None)  # select all the track
+
+            # compute 2D density
+            hist_new, _, _ = np.histogram2d(
+                track.lat.values[index],
+                track.lon.values[index],
+                bins=[lat_bins, lon_bins],
+                density=False,
+            )
+            if filter_tracks:
+                hist_new[hist_new > 1] = 1
+
+            hist_count += hist_new
+
+    if norm:
+        hist_count = normalize_hist(res=res, hist_count=hist_count, norm=norm)
+
+    return hist_count, lat_bins, lon_bins
+
+
+def compute_genesis_density(
+    tc_track: TCTracks, lat_bins: np.ndarray, lon_bins: np.ndarray
+) -> np.ndarray:
+    """Compute the density of track genesis locations. This function works under the hood
+    of :py:meth:`compute_track_density`. If it is called with the parameter genesis = True,
+    the function return the number of genesis points per grid cell.
+
+    Parameters:
+    -----------
+
+    tc_track: TCT track object
+        TC track object containing a list of all tracks.
+    lat_bins: 1D np.array
+        array containg the latitude bins.
+    lon_bins: 1D np.array
+        array containg the longitude bins.
+
+    Returns:
+    --------
+    hist_count: 2D np.array
+        array containing the number of genesis points per grid cell
+    """
+    # Extract the first lat and lon from each dataset
+    first_lats = np.array([ds.lat.values[0] for ds in tc_track.data])
+    first_lons = np.array([ds.lon.values[0] for ds in tc_track.data])
+
+    # compute 2D density of genesis points
+    hist_count, _, _ = np.histogram2d(
+        first_lats,
+        first_lons,
+        bins=[lat_bins, lon_bins],
+        density=False,
+    )
+
+    return hist_count
+
+
+def normalize_hist(
```

Comment:
Is this supposed to be a method called by the user? If yes, leave it as is. If not, please make it private with the `_`.